modules/common/src/main/scala/higherkindness/mu/rpc/codegen/ServiceGenerator.scala

```diff
@@ -0,0 +1,720 @@
+
+      private def clientCallMethodFor(clientMethodName: String) =
+        q"$clientCallsImpl.${TermName(clientMethodName)}(input, $methodDescriptorName, channel, options)"
+
+      val clientDef: Tree = streamingType match {
+        case Some(RequestStreaming) =>
+          q"""
+            def $name(input: ${request.getTpe}): ${response.getTpe} = ${clientCallMethodFor(
+            "clientStreaming")}"""
+        case Some(ResponseStreaming) =>
+          q"""
+            def $name(input: ${request.getTpe}): ${response.getTpe} = ${clientCallMethodFor(
+            "serverStreaming")}"""
+        case Some(BidirectionalStreaming) =>
+          q"""
+            def $name(input: ${request.getTpe}): ${response.getTpe} = ${clientCallMethodFor(
+            "bidiStreaming")}"""
+        case None =>
+          q"""
+            def $name(input: ${request.getTpe}): ${response.getTpe} = ${clientCallMethodFor("unary")}"""
+      }
+
+      private def serverCallMethodFor(serverMethodName: String) =
+        q"_root_.higherkindness.mu.rpc.internal.server.monixCalls.${TermName(serverMethodName)}(algebra.$name, $compressionOption)"
+
+      val descriptorAndHandler: Tree = {
+        val handler = (streamingType, prevalentStreamingTarget) match {
+          case (Some(RequestStreaming), Fs2StreamTpe(_, _)) =>
+            q"_root_.higherkindness.mu.rpc.internal.server.fs2Calls.clientStreamingMethod(algebra.$name, $compressionOption)"
+          case (Some(RequestStreaming), MonixObservableTpe(_, _)) =>
+            q"_root_.io.grpc.stub.ServerCalls.asyncClientStreamingCall(${serverCallMethodFor("clientStreamingMethod")})"
+          case (Some(ResponseStreaming), Fs2StreamTpe(_, _)) =>
+            q"_root_.higherkindness.mu.rpc.internal.server.fs2Calls.serverStreamingMethod(algebra.$name, $compressionOption)"
+          case (Some(ResponseStreaming), MonixObservableTpe(_, _)) =>
+            q"_root_.io.grpc.stub.ServerCalls.asyncServerStreamingCall(${serverCallMethodFor("serverStreamingMethod")})"
+          case (Some(BidirectionalStreaming), Fs2StreamTpe(_, _)) =>
+            q"_root_.higherkindness.mu.rpc.internal.server.fs2Calls.bidiStreamingMethod(algebra.$name, $compressionOption)"
+          case (Some(BidirectionalStreaming), MonixObservableTpe(_, _)) =>
+            q"_root_.io.grpc.stub.ServerCalls.asyncBidiStreamingCall(${serverCallMethodFor("bidiStreamingMethod")})"
+          case (None, _) =>
+            q"_root_.io.grpc.stub.ServerCalls.asyncUnaryCall(_root_.higherkindness.mu.rpc.internal.server.unaryCalls.unaryMethod(algebra.$name, $compressionOption))"
+          case _ =>
+            sys.error(
+              s"Unable to define a handler for the streaming type $streamingType and $prevalentStreamingTarget for the method $name in the service $serviceName")
+        }
+        q"($methodDescriptorName, $handler)"
+      }
+    }
+
+    case class HttpOperation(operation: Operation) {
```

Comment:
> we'll want to split this into client and server classes for better readability.

Would it make sense to split the Http part too?

Reply:
Absolutely! Currently we have a monolithic straight-up copy of the current macro output but there's lots of improvements we can make so it becomes more user-friendly.

Also moving off macros might help resolve the issue that led us to put the HTTP stuff in there in the first place, where we couldn't get two processors for the same macro applied from different classes.